Plugins/NativeCamera/Include/Babylon/Plugins/NativeCamera.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <napi/env.h>
+
+namespace Babylon::Plugins
+{
+    class Camera final
+    {
+    public:
+        class Impl;
+
+        Camera(const Camera& other) = default;
+        Camera(Camera&&) = default;
+        ~Camera() = default;
+
+        // Initialization with overrideCameraTexture set to true means the caller
+        // is expected to override the camera source texture with a native texture.
+        // In that case, the HW camera will not be open.
+        static Camera Initialize(Napi::Env env, bool overrideCameraTexture = false);
+
+        void SetTextureOverride(void* texturePtr);
```

Comment:
Can we make this strongly typed instead of `void*`? Like add a `using TextureType = int` to `GraphicsPlatform.h` for Android for example and then use that header here?

Reply:
I agree. I will do that in a second PR once the GraphicsContext PR is in as well.

Reply:
This proved to be more difficult than anticipated. Using Apple types (id<texture>) in .h/.cpp brings a lot of build issues. I think a 3rd (or 4th) refactor is needed. I will do it later.